config/v1/types_infrastructure.go

```diff
@@ -51,28 +56,47 @@ type InfrastructureStatus struct {
 type PlatformType string
 
 const (
-	// AWSPlatform represents Amazon Web Services infrastructure.
-	AWSPlatform PlatformType = "AWS"
+	// AWSPlatformType represents Amazon Web Services infrastructure.
+	AWSPlatformType PlatformType = "AWS"
 
-	// AzurePlatform represents Microsoft Azure infrastructure.
-	AzurePlatform PlatformType = "Azure"
+	// AzurePlatformType represents Microsoft Azure infrastructure.
+	AzurePlatformType PlatformType = "Azure"
 
-	// GCPPlatform represents Google Cloud Platform infrastructure.
-	GCPPlatform PlatformType = "GCP"
+	// GCPPlatformType represents Google Cloud Platform infrastructure.
+	GCPPlatformType PlatformType = "GCP"
 
-	// LibvirtPlatform represents libvirt infrastructure.
-	LibvirtPlatform PlatformType = "Libvirt"
+	// LibvirtPlatformType represents libvirt infrastructure.
+	LibvirtPlatformType PlatformType = "Libvirt"
 
-	// OpenStackPlatform represents OpenStack infrastructure.
-	OpenStackPlatform PlatformType = "OpenStack"
+	// OpenStackPlatformType represents OpenStack infrastructure.
+	OpenStackPlatformType PlatformType = "OpenStack"
 
-	// NonePlatform means there is no infrastructure provider.
-	NonePlatform PlatformType = "None"
+	// NonePlatformType means there is no infrastructure provider.
+	NonePlatformType PlatformType = "None"
 
-	// VSpherePlatform represents VMWare vSphere infrastructure.
-	VSpherePlatform PlatformType = "VSphere"
+	// VSpherePlatformType represents VMWare vSphere infrastructure.
+	VSpherePlatformType PlatformType = "VSphere"
 )
 
+// PlatformSpec contains settings that are specific to every underlying infrastructure
+// provider supported by these specs. Since these are used at spec-level, it is a valid
+// scenario to have attributes in more than one platform provider spec.
+type PlatformSpec struct {
+	// AWSPlatform contains settings specific to the Amazon Web Services infrastructure provider.
+	// +optional
+	AWSPlatform AWSPlatformSpec `json:"awsPlatform,omitempty"`
+}
+
+// AWSPlatformSpec specify settings specific to the Amazon Web Services infrastructure provider.
+type AWSPlatformSpec struct {
+	// infraID uniquely identifies an AWS cluster with a human friendly ID. must be of max length
+	// 27 and must have only alphanumeric or hyphen characters.
+	InfraID string `json:"infraID"`
```

Comment:
Don't shorten names - `infrastructureName` (a name is an identifier that is human friendly).